lib/config/createUrc.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const path = require('path');
+const _ = require('lodash');
+const { WEBPACK_ASSETS_BASENAME } = require('../constants');
+const validateConfig = require('./validate-config');
+const { dynamicRequireSync } = require('../utils/dynamicRequire');
+
+module.exports = createUrc;
+
+class Urc {
+  constructor(rawConfig = {}, defaultConfig) {
+    validateConfig(rawConfig);
+    const urc = _.merge({}, defaultConfig, rawConfig);
+    // Normalize URL parts.
+    if (urc.siteBasePath && urc.siteBasePath !== '/') {
+      urc.siteBasePath = urc.siteBasePath.replace(/\/$/, '');
+      if (!urc.siteBasePath.startsWith('/')) {
+        urc.siteBasePath = '/' + urc.siteBasePath;
+      }
+    }
+    if (!urc.publicAssetsPath.startsWith('/')) {
+      urc.publicAssetsPath = '/' + urc.publicAssetsPath;
+    }
+    Object.assign(this, urc);
+  }
+
+  getAssets() {
+    return dynamicRequireSync({
+      path: path.join(this.outputDirectory, WEBPACK_ASSETS_BASENAME),
+      deleteCache: true
+    });
+  }
+
+  getHtmlSourceFn() {
+    return dynamicRequireSync({
+      path: this.htmlSource,
+      backupPath: path.join(__dirname, '..', './default-html.js'),
```

Comment:
Could `htmlSource` just have a default? Would that avoid the need for this `backupPath` option?

Reply:
Ya that makes sense, I have made the default htmlSource in `default-underreact.config.js` point in the latest [commit](https://github.com/mapbox/underreact/pull/13/commits/d4e9968ca0635909081a894266883a9a93095a51#diff-9641d65535017283a8bbb631f8f5a48cR33)